### PR TITLE
First Nice PR :)

### PR DESCRIPTION
**Pull request by SavaVuckovic**
Refactoring the code to fix issues pointed out in issue #1

### CODE REVIEW

**Comment by nidalaa** [comment]
Another PR! So much happiness! 😍 

awesome

File: spec/enumerable_extension_spec.rb [hunk 1]
```diff
@@ -2,132 +2,107 @@
...
 
-      expect(actual).to eq(expected)
+      expect(result).to eq([2,3,4,5])
     end
 
   end
 
   describe "#my_each_with_index" do
 
     it "doesn't modify original array" do
```
Comment (nidalaa) [inline spec/enumerable_extension_spec.rb:26]: I think my_each_with_index should also repeat this spec from my_each https://github.com/talihomz/Enumaruby/pull/2/files#diff-95d934febaaa9135d9c567171be87643R15

As `my_each_with_index` does exactly the same thing as `my_each` and in addition handles indices.

**Review by nidalaa — COMMENTED** [review]
One comment. Other than that well done!

**Review by nidalaa — CHANGES_REQUESTED** [review]
AS in comment above ☝️

**Review by talihomz — APPROVED** [review]
Good work Sava ^^

**Comment by nidalaa** [comment]
@wahomekevin @SavaVuckovic why was it merged without a final 👍 from the reviewer? 😨

**Comment by talihomz** [comment]
I thought we had finalized everything on this?

Also, since it is already done, what is the usual process for undoing such a review, and restoring to the original state?

**Comment by talihomz** [comment]
I've gone ahead and clicked on the "revert" button. Is this the right way?

**Pull request merged**